openpype/tools/ayon_workfiles/widgets/files_widget_workarea.py

```diff
@@ -75,16 +75,18 @@ def _get_missing_context_item(self):
             self._missing_context_item = item
         return self._missing_context_item
 
-    def clear(self):
-        self._items_by_filename = {}
+    def _clear(self):
         self._remove_missing_context_item()
         self._remove_empty_item()
-        super(WorkAreaFilesModel, self).clear()
+        if self._items_by_filename:
+            root = self.invisibleRootItem()
+            root.removeRows(0, root.rowCount())
+            self._items_by_filename = {}
```

Comment:
Why does this fix column count? :)

Reply:
If you call `self.clear()` it will change `columnCount` to `0`.

Reply:
And if so, what issues would that give? And why does `removeRows` which I suppose removes ALL rows solve that?
Since you have a custom module - why not override the `columnCount` to always return the amount you need.
This feels like a fix for a problem elsewhere.